pkg/receive/multitsdb.go

```diff
@@ -493,6 +498,29 @@ func (t *MultiTSDB) startTSDB(logger log.Logger, tenantID string, tenant *tenant
 	reg = NewUnRegisterer(reg)
 
 	lset := labelpb.ExtendSortedLabels(t.labels, labels.FromStrings(t.tenantLabelName, tenantID))
+
+hcLoop:
+	for _, hc := range t.hashringConfigs {
+		for _, t := range hc.Tenants {
+			if t == tenantID {
+				var elset labels.Labels
+				for _, el := range hc.ExternalLabels {
+					parts := strings.SplitN(el, "=", 2)
+					if len(parts) != 2 {
+						return errors.Errorf("unrecognized label %q", el)
+					}
+					if !model.LabelName.IsValid(model.LabelName(parts[0])) {
+						return errors.Errorf("unsupported format for label %s", el)
+					}
+					elset = append(elset, labels.Label{Name: parts[0], Value: parts[1]})
+				}
+				sort.Sort(elset)
+				lset = labelpb.ExtendSortedLabels(lset, elset)
+				break hcLoop
+			}
+		}
+	}
```

Comment:
@fpetkovski You mentioned extracting external labels above. Actually right now here's where we extract and handle external labels from hashring configs.

Could you please review if this logic works for both adding and handling changes for external labels? For adding maybe yes because adding was checked successfully locally. But for handling changes, I'm not quite sure.

The problem now is external labels can be changed successfully in `MultiTSDB`. But the changes don't show up in query. I'm still figuring out why.